label_studio/data_manager/models.py

```diff
@@ -86,7 +86,25 @@ class FilterGroup(models.Model):
 
 
 class Filter(models.Model):
-    index = models.IntegerField(_('index'), default=0, help_text='To keep filter order')
+    # Optional reference to a parent filter. We only allow **one** level of nesting.
+    parent = models.ForeignKey(
+        'self',
+        on_delete=models.CASCADE,
+        related_name='children',
+        null=True,
+        blank=True,
+        db_index=False,
+        help_text='Optional parent filter to create one-level hierarchy (child filters are AND-merged with parent)',
+    )
+
+    # `index` is now only meaningful for **root** filters (parent is NULL)
+    index = models.IntegerField(
```

Comment:
is there a way we could name this field to make it clearer what it really represents? `display_index` or `display_order_index` would both help in this regard, imo

Reply:
This field isn't new, renaming it would have increased change surface area, but yes it's possible. Could do it in a separate chore PR